torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -9722,6 +9722,39 @@ def forward(self, inp):
             )
             model_ddp2(input).sum().backward()
 
+        @skip_if_lt_x_gpu(2)
+        @skip_but_pass_in_sandcastle_if(
+            BACKEND not in DistTestCases.backend_feature["ddp"],
+            f"The {BACKEND} backend does not support DistributedDataParallel",
+        )
+        def test_ddp_has_finalized(self):
+            class MyModel(nn.Module):
+                def __init__(self, device):
+                    super(MyModel, self).__init__()
+                    self.error = True
+                    self.fc1 = nn.Linear(10, 10).cuda(device)
+                    self.fc2 = nn.Linear(10, 10).cuda(device)
+
+                def forward(self, inp):
+                    return self.fc1(inp), self.fc2(inp)
+
+            model = MyModel(self.rank)
+            input = torch.rand(10, 10, requires_grad=True).cuda(self.rank)
+            ddp = torch.nn.parallel.DistributedDataParallel(
+                model,
+                device_ids=[self.rank],
+            )
+
+            out1, _ = ddp(input)
+            out1.sum().backward()
+
+            with self.assertRaisesRegex(RuntimeError, "Reducer has not been finalized!"):
```

Comment:
if this is run with find_unused_parameters=True or static_graph (and the graph is static) then I believe this training should work. In that case, will _check_finalized return true or false?

Reply:
Good point, the issue I ran into was with `find_unused_parameters=True`. Let me update the unit test with `find_unused_parameters` to illustrate the problem.

Reply:
@rohan-varma Updated the test